taichi/backends/metal/kernel_manager.cpp

```diff
@@ -580,7 +579,7 @@ class KernelManager::Impl {
         compiled_structs_.runtime_size + mem_pool_bytes, mem_pool_);
     runtime_buffer_ = new_mtl_buffer_no_copy(device_.get(), runtime_mem_->ptr(),
                                              runtime_mem_->size());
-    buffer_meta_data_.runtime_buffer_size = runtime_mem_->size();
+    buffer_meta_data_.runtime_buffer_size = compiled_structs_.runtime_size;
```

Comment:
```suggestion
    // Ignore |mem_pool_bytes| because memory pool is only used when there are sparse SNodes.
    buffer_meta_data_.runtime_buffer_size = compiled_structs_.runtime_size;
```